samples/apps/copilot-chat-app/webapi/CopilotChat/Skills/ChatSkills/ChatSkill.cs

```diff
@@ -158,9 +159,26 @@ public async Task<string> ExtractChatHistoryAsync(SKContext context)
             var formattedMessage = chatMessage.ToFormattedString();
             var tokenCount = Utilities.TokenCount(formattedMessage);
 
-            // Plan object is not meaningful content in generating chat response, exclude it
-            if (remainingToken - tokenCount > 0 && !formattedMessage.Contains("proposedPlan\\\":", StringComparison.InvariantCultureIgnoreCase))
+            if (remainingToken - tokenCount > 0)
             {
+                // Plan object is not meaningful content in generating chat response, reduce to intent to save on tokens
+                if (formattedMessage.Contains("proposedPlan\":", StringComparison.InvariantCultureIgnoreCase))
+                {
+                    string pattern = @"(\[.*?\]).*User Intent:User intent: (.*)(?=""}})";
+                    Match match = Regex.Match(formattedMessage, pattern);
+                    if (match.Success)
+                    {
+                        string timestamp = match.Groups[1].Value.Trim();
+                        string userIntent = match.Groups[2].Value.Trim();
+
+                        formattedMessage = $"{timestamp} Bot proposed plan to fulfill user intent: {userIntent}";
```

Comment:
Shouldn't we update the token count here and after line 178?

Reply:
Moved the plan object truncation out of this if block before tokenCount is calculated